src/main/msc/emfat_file.c

```diff
@@ -291,44 +294,102 @@ static void emfat_add_log(emfat_entry_t *entry, int number, uint32_t offset, uin
     entry->curr_size = size;
     entry->max_size = entry->curr_size;
     entry->readcb = bblog_read_proc;
-    emfat_set_entry_cma(entry);
+    // Set file modification/access times to be the same, either the default date or that from the RTC
+    entry->cma_time[1] = cmaTime;
+    entry->cma_time[2] = cmaTime;
```

Comment:
This section should not be changed. Initially setting all the dates to the default is what we want.

This causes a problem if your parsing doesn't find a date as it gets left as the "0" date.

Reply:
There are three dates to consider, creation, modification and access. The creation date is what I'm extracting from the log as that is indeed the creation date. If there is no date in the log this is detected at line 354 (zero year) in which case the default is applied, which typically is the time at which the flash is mounted. So, no issue with the "0" date. cmaTime is either 1/1/2018 @ 1pm or the current time passed to the FC over USB.

Modification time should ideally be set to the time that the log is closed which we don't know easily, and I'm not sure that parsing the log to figure this out brings much benefit. There's an argument (having gone for simplicity!) for setting this to either the creation time or the time when the flash is mounted for read. I've picked the latter, but can easily make this the former. Similar argument for access really.

All I was doing above was leaving the modification/access dates as they were before.

I'll change them both to the creation time.

Reply:
The point is I tested with a log the intentionally had no `Log start datetime:` header to make the parsing fail. In that case the creation date/time was set to 0.

Reply:
And now the dates are all set to the same value.

Just tested with a log with no date and it has all three timestamps set to the time the filesystem was mounted.